smokes_mojo/lib/SmokesMojo/Controller/Api.pm

```diff
@@ -85,6 +88,58 @@ $msg
 EOS
 }
 
+sub post_coresmokedb_report ($self) {
+    my $req = $self->req;
+
+    my $schema = SmokeReports::Dbh->schema;
+    my $sdb = $schema->resultset("Perl5Smoke");
+    my $id = $sdb->get_column("report_id")->max;
+    $id = $id + 1;
```

Comment:
If this PR is intended to make perl.develop-help.com the only submission point for reports, this is unsafe, since reports from two different smokes could be received at close to the same time.

perl.develop-help.com uses the prefork server, so if I understand correctly it can be processing two submissions in parallel, in which case this could allocate the same id to both reports.

The only way I could see it working would be to make report_id autoincrement, which would require some other changes to the table (and I'd need to check nothing refers to records by id, I forget)

If it will be accepting reports in parallel with perl5.test-smoke.org things become more complex.